tests/Unit/NumericalAlgorithms/DiscontinuousGalerkin/BoundarySchemes/StrongFirstOrder/Test_BoundaryFlux.cpp

```diff
@@ -0,0 +1,255 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/SliceIterator.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "Domain/FaceNormal.hpp"
+#include "Domain/InterfaceComputeTags.hpp"
+#include "Domain/InterfaceHelpers.hpp"
+#include "Domain/Mesh.hpp"
+#include "Domain/Tags.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/BoundarySchemes/StrongFirstOrder/BoundaryData.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/BoundarySchemes/StrongFirstOrder/BoundaryFlux.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/MortarHelpers.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Protocols.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/MakeString.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "tests/Unit/ProtocolTestHelpers.hpp"
+#include "tests/Unit/Pypp/CheckWithRandomValues.hpp"
+#include "tests/Unit/Pypp/SetupLocalPythonEnvironment.hpp"
+#include "tests/Utilities/MakeWithRandomValues.hpp"
+
+namespace {
+
+struct SomeField : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+
+struct ExtraDataTag : db::SimpleTag {
+  using type = int;
+};
+
+struct NumericalFlux : tt::ConformsTo<dg::protocols::NumericalFlux> {
+  using variables_tags = tmpl::list<SomeField>;
+  using argument_tags = tmpl::list<SomeField, ExtraDataTag>;
+  using volume_tags = tmpl::list<ExtraDataTag>;
+  using package_field_tags = tmpl::list<SomeField>;
+  using package_extra_tags = tmpl::list<ExtraDataTag>;
+  static void package_data(
+      const gsl::not_null<Scalar<DataVector>*> packaged_field,
+      const gsl::not_null<int*> packaged_extra_data,
+      const Scalar<DataVector>& field, const int& extra_data) noexcept {
+    *packaged_field = field;
+    *packaged_extra_data = extra_data;
+  }
+  void operator()(const gsl::not_null<Scalar<DataVector>*> numerical_flux,
+                  const Scalar<DataVector>& field_int,
+                  const int& extra_data_int,
+                  const Scalar<DataVector>& field_ext,
+                  const int& extra_data_ext) const noexcept {
+    CHECK(extra_data_int == extra_data_ext);
+    // A simple central flux
+    get(*numerical_flux) = 0.5 * (get(field_int) + get(field_ext));
+  }
+};
+
+static_assert(
+    test_protocol_conformance<NumericalFlux, dg::protocols::NumericalFlux>,
+    "Failed testing protocol conformance");
+
+// A flux used in earlier versions of this test (see history of
+// `Test_MortarHelpers.cpp`), included here to make sure it didn't break
```

Comment:
Use a full path if you want to reference file history so it can be found if the file is removed or renamed.